### PR TITLE
fix: [slider] remove focus management

### PR DESCRIPTION
**Pull request by eljefe223**
# Pull Request

## 📖 Description
This change removes the focus management from `slider` because its not needed and it makes assumptions about how `:focus-visible` should be rendered.

Calling `.focus()`  will always force the component in a `:focus-visible` state. Removing these calls allows for more flexibility with focus styling.

### General

- [x] I have included a change request file using `$ yarn change`
- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [ x] I have read the [CONTRIBUTING](https://github.com/microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](/docs/community/code-of-conduct/#our-standards) for this project.

### Component-specific

- [ ] I have added a new component
- [x] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

### CODE REVIEW

File: change/@microsoft-fast-foundation-848c548a-723d-4657-9096-b9896e9d2b73.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-848c548a-723d-4657-9096-b9896e9d2b73.json:2]:
```suggestion
  "type": "prerelease",
```

File: change/@microsoft-fast-foundation-848c548a-723d-4657-9096-b9896e9d2b73.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+  "comment": "fix: remove focus management from slider",
+  "packageName": "@microsoft/fast-foundation",
+  "email": "jes@microsoft.com",
+  "dependentChangeType": "patch"
```
Comment (chrisdholt) [inline change/@microsoft-fast-foundation-848c548a-723d-4657-9096-b9896e9d2b73.json:6]:
```suggestion
  "dependentChangeType": "prerelease"
```

**Review by chrisdholt — COMMENTED** [review]
I want to pull this down to ensure that we aren't missing something obvious, but in the meantime let's be sure we target prerelease (master is tracking alpha/beta versions)

**Comment by chrisdholt** [comment]
@eljefe223 thanks for this fix - I think this makes sense. Focus in this case is _always_ on the host as it is the element with a role of slider. If someone wants to style the thumb they can do so accordingly, but I don't think we should move to an internal element which has no role.

**Review by chrisdholt — APPROVED** [review]

File: packages/web-components/fast-foundation/src/slider/slider.ts [hunk 3]
```diff
@@ -410,12 +410,6 @@ export class FASTSlider extends FormAssociatedSlider implements SliderConfigurat
      *  If the event handler is null it removes the events
      */
     private handleThumbMouseDown = (event: MouseEvent | null): void => {
-        if (event) {
-            if (this.readOnly || this.disabled || event.defaultPrevented) {
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/slider/slider.ts:414]: @eljefe223 One catch - we should probably keep the check here and return if it's readonly/disabled/etc. We should not focus, but I think this guard is good.

File: packages/web-components/fast-foundation/src/slider/slider.ts [hunk 4]
```diff
@@ -496,9 +490,7 @@ export class FASTSlider extends FormAssociatedSlider implements SliderConfigurat
             window[eventAction]("mousemove", this.handleMouseMove);
 
             if (e) {
-                e.preventDefault();
```
Comment (chrisdholt) [inline packages/web-components/fast-foundation/src/slider/slider.ts:499]: Do we need to remove this?

**Review by chrisdholt — CHANGES_REQUESTED** [review]
Found one change and one question

File: packages/web-components/fast-foundation/src/slider/slider.ts (hunk 4 above)
Comment (eljefe223) [reply packages/web-components/fast-foundation/src/slider/slider.ts:499]: Yes this is preventing the focus naturally going to `slider`

File: packages/web-components/fast-foundation/src/slider/slider.ts (hunk 3 above)
Comment (eljefe223) [reply packages/web-components/fast-foundation/src/slider/slider.ts:414]: handleMouseMove and handleMouseDown already catch disabled which are the events this method is adding, I think if we want to be safe we can add the disabled check alongside `readOnly` on line 293 in handleKeyDown. I think we had redundant checking.

**Review by chrisdholt — APPROVED** [review]

**Review by nicholasrice — APPROVED** [review]

**Pull request merged**